inference-engine/src/mkldnn_plugin/nodes/mkldnn_pooling_node.cpp

```diff
@@ -231,26 +247,45 @@ void MKLDNNPoolingNode::prepareParams() {
         if (auto_pad) {
             std::tie(data_pad_begin, data_pad_end) = getPaddingFromNode(opToShapeInfer);
         }
-        initEffectivePad(inDesc->getShape(), outDesc->getShape());
+        initEffectiveAttributes(inDesc->getShape(), outDesc->getShape());
     }
 
-    mkldnn::algorithm alg = getPoolingAlgorithm();
-    MKLDNNDescriptor desc{createDescriptorInternal(in_candidate, out_candidate, alg)};
-    pooling_forward::primitive_desc prim_desc;
-    primitive_desc_iterator itpd = desc.createPrimitiveDescriptorIterator(getEngine(), *attr);
+    if (isMaxPool8) {
+        MKLDNNDescriptor desc{createDescriptorInternalV2(in_candidate, out_candidate)};
+        pooling_v2_forward::primitive_desc prim_desc;
+        primitive_desc_iterator itpd = desc.createPrimitiveDescriptorIterator(getEngine(), *attr);
 
-    while (static_cast<bool>(itpd))  {
-        impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
+        while (static_cast<bool>(itpd)) {
+            impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
 
-        if (impl_type == selected_pd->getImplementationType()) {
-            prim_desc = itpd.get();
-            break;
+            if (impl_type == selected_pd->getImplementationType()) {
+                prim_desc = itpd.get();
+                break;
+            }
+            if (!itpd.next_impl())
+                IE_THROW() << "Primitive descriptor was not found for node " << getName() << ".";
         }
-        if (!itpd.next_impl())
-            IE_THROW() << "Primitive descriptor was not found for node " << getName() << ".";
-    }
 
-    prim.reset(new pooling_forward(prim_desc));
+        prim.reset(new pooling_v2_forward(prim_desc));
```

Comment:
Why cannot we use pooling_v2_forward in all cases and get rid of legacy version?